gta/map_structures.py

```diff
@@ -0,0 +1,162 @@
+SA_structures['inst'] =  namedtuple("IPL_INST_SA", "id modelName interior posX posY posZ rotX rotY rotZ rotW lod")
+
+# CULL
+# Creates zones with special attributes
+III_structures['cull'] = namedtuple("IPL_CULL_3",      "centerX centerY centerZ lowerLeftX lowerLeftY lowerLeftZ upperRightX upperRightY upperRightZ flags wantedLevelDrop")
+VC_structures['cull'] =  namedtuple("IPL_CULL_VC",     "centerX centerY centerZ lowerLeftX lowerLeftY lowerLeftZ upperRightX upperRightY upperRightZ flags wantedLevelDrop")
+SA_structures['cull_1'] =  namedtuple("IPL_CULL_SA_1", "centerX centerY centerZ unknown1 widthY bottomZ widthX unknown2 topZ flag unknown3")
+SA_structures['cull_2'] =  namedtuple("IPL_CULL_SA_2", "centerX centerY centerZ unknown1 widthY bottomZ widthX unknown2 topZ flag Vx Vy Vz cm")
+
+# ZONE
+# Creates map, navigation, and info zones
+III_structures['zone'] = namedtuple("IPL_ZONE_3",  "name type x1 y1 z1 x2 y2 z2 level")
+VC_structures['zone'] =  namedtuple("IPL_ZONE_VC", "name type x1 y1 z1 x2 y2 z2 level")
+SA_structures['zone'] =  namedtuple("IPL_ZONE_SA", "name type x1 y1 z1 x2 y2 z2 level")
+
+# GRGE
+# Creates a garage
+SA_structures['grge'] = namedtuple("IPL_GRGE_SA", "posX posY posZ lineX lineY cubeX cubeY cubeZ doorType garageType name")
+
+# ENEX
+# Creates an entrance to an exit
+SA_structures['enex'] = namedtuple("IPL_ENEX_SA", "x1 y1 z1 enterAngle sizeX sizeY sizeZ x2 y2 z2 exitAngle targetInterior flags name sky numPedsToSpawn timeOn timeOff")
+
+# PICK
+# Creates pickups. This section exists in GTA III, GTA Vice City, and GTA IV, but is only functional in GTA San Andreas
+SA_structures['pick'] = namedtuple("IPL_PICK_SA", "id posX posY posZ")
+
+# JUMP
+# Creates a unique stunt jump
+SA_structures['jump'] = namedtuple("IPL_JUMP_SA", "startLowerX startLowerY startLowerZ startUpperX startUpperY startUpperZ targetLowerX targetLowerY targetLowerZ targetUpperX targetUpperY targetUpperZ cameraX cameraY cameraZ reward")
+
+# AUZO
+# This creates an audio if you enter the zone
+SA_structures['auzo_1'] = namedtuple("IPL_AUZO_SA_1", "name id switch x y z radius")
+SA_structures['auzo_2'] = namedtuple("IPL_AUZO_SA_2", "name id switch x1 y1 z1 x2 y2 z2")
+
+# CARS
+# Creates a parked car generator
+SA_structures['cars'] = namedtuple("IPL_CARS_SA", "posX posY posZ angle carId primCol secCol forceSpawn alarm doorLock unknown1 unknown2")
+
+# OCCL
+# Creates occlusion zones for separated rendering
+VC_structures['occl'] = namedtuple("IPL_OCCL_VC", "midX midY bottomZ widthX widthY height rotation")
+SA_structures['occl'] = namedtuple("IPL_OCCL_SA", "midX midY bottomZ widthX widthY height rotation")
+
+#############
+#    IDE    #
+#############
+
+# Sources
+# https://www.grandtheftwiki.com/Item_Definition
+# https://gta.fandom.com/wiki/IDE
+
+# Not present
+# PATH
+# WEAP - couldn't confirm format, or even supporting games
```

Comment:
Supports GTA VC - GTA LCS
`Id, ModelName, TxdName, AnimationName, MeshCount, DrawDistance`

Source: [https://gtamods.com/wiki/WEAP](https://gtamods.com/wiki/WEAP)

Reply:
If you're sure it's VC and LCS I can put it in...
But https://gta.fandom.com/wiki/IDE claims it's a III and VC format
and https://www.grandtheftwiki.com/WEAP claims it's a VC and SA format.

You understand my confusion. 😄

Reply:
I believe it is so. :stuck_out_tongue: 
And yeah, GTA VC - LCS means GTA VC to LCS. This includes San Andreas, sorry for the confusion.

P.S gtamods.com is an accurate source of documentation for GTA format. In fact, the articles related to game file documentation in GTA Fandom and GrandTheftWiki have mostly been imported from GTAMods (formerly gtamodding).